### ISSUE TITLE
Wrong criticality when first nuclide in material has density of 0

### ISSUE THREAD

**Issue by ohnemax** [openmc-dev/openmc#1220]
This is a very specific error, and will very likely not affect many people. And probably it is an example for a wrongly written input file. However, it affected me and it took me some time to figure out what was going on, so I post it here.

When the first nuclide in a material has a density set to 0, criticality results show a very erratic behavior. A working example is attached.

Basically, the following material composition yields a combined k-effective of 0.76571 +/-  0.00025:
```
<material id="1" name="1.6% Fuel">
    <temperature>900</temperature>
    <density units="g/cm3" value="10.31341" />
    <nuclide ao="0.0" name="U234" />
    <nuclide ao="0.00037503" name="U235" />
    <nuclide ao="0.022625" name="U238" />
    <nuclide ao="0.046007" name="O16" />
  </material>
```

However, if the line with ao="0.0" for U234 is removed, it works fine. In this case criticality is 1.01150 +/-  0.00024. If the line is moved somewhere after a nuclide that has a non-zero density, it works fine, too. (criticality 1.01150).

I tested this with [v0.10.0](https://github.com/openmc-dev/openmc/releases/tag/v0.10.0), compiled for cluster usage using OpenMPI 2.0.2. It occurs both for ENDF-B VII.0 and ENDF-B VIII cross section data sets. 

**Example Files**
[Example with wrong result](https://github.com/openmc-dev/openmc/files/3071030/example-endfb-VII.0.zip)
[Example with correct result](https://github.com/openmc-dev/openmc/files/3071031/example-endfb-VII.0-working.zip)

**Comment by paulromano** [comment]
Thanks for reporting this @ohnemax. Yes, this is kind of a corner case that I don't expect most people to run into. Nevertheless, I've submitted a bugfix for this in #1219.

**Comment by paulromano** [comment]
Fixed with #1219
